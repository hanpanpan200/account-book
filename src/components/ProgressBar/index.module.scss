@import "../../styles/colors";

.container {
  height: 5px;
  width: 150px;
  position: relative;
  top: 6px;
  border-radius: 2px;
  background-color: #d9d9d9;

  div {
    height: 5px;
    background-color: $red;
    border-radius: 2px;
    min-width: 1px;
  }
}