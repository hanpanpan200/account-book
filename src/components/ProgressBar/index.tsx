import React from 'react';

import styles from './index.module.scss';

interface Props {
  percentage: number
}

const ProgressBar:React.FC<Props> = ({ percentage }) => {
  return (
    <div className={styles.container}>
      <div style={{ width: `${percentage}%` }} />
    </div>
  )
}

export default ProgressBar;