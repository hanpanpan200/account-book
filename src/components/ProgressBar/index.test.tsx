import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from './index';

describe('ProgressBar', () => {
  it('should render correctly', () => {
    const { container } = render(<ProgressBar percentage={80} />);
    expect(container).toMatchSnapshot();
  });
});
