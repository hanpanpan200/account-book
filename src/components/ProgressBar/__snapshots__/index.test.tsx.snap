// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      style="width: 80%;"
    />
  </div>
</div>
`;
